base/bitarray.jl

```diff
@@ -1779,26 +1779,40 @@ end
 # map across the chunks. Otherwise, fall-back to the AbstractArray method that
 # iterates bit-by-bit.
 function bit_map!(f::F, dest::BitArray, A::BitArray) where F
-    size(A) == size(dest) || throw(DimensionMismatch("sizes of dest and A must match"))
+    length(A) <= length(dest) || throw(DimensionMismatch("length of dest must be >= than length of A"))
     isempty(A) && return dest
     destc = dest.chunks
     Ac = A.chunks
-    for i = 1:(length(Ac)-1)
+    len_Ac = length(Ac)
+    for i = 1:(len_Ac-1)
         destc[i] = f(Ac[i])
     end
-    destc[end] = f(Ac[end]) & _msk_end(A)
+    # the last effected UInt64's original content
+    dest_last = destc[len_Ac]
+    _msk = _msk_end(A)
+    # first zero out the bits mask is going to change
+    destc[len_Ac] = (dest_last & ~(_msk))
+    # then update bits by `or`ing with a masked RHS
+    destc[len_Ac] |= f(Ac[end]) & _msk
     dest
 end
 function bit_map!(f::F, dest::BitArray, A::BitArray, B::BitArray) where F
-    size(A) == size(B) == size(dest) || throw(DimensionMismatch("sizes of dest, A, and B must all match"))
+    length(A) == length(B) || throw(DimensionMismatch("size of A and B must be equal"))
```

Comment:
I think we can loose this restriction a bit.
Just set `len_Ac = min(length(Ac), length(BC))` and `_msk = _msk_end(len_Ac)`

Reply:
I don't want people to silently feed the wrong arrays though, at least this is looser than before

Reply:
But that matches the general behavior of `map(!)`.
We can always call `@invoke map!(f, dest::AbstractArray, A::AbstractArray, B::AbstractArray)` in this case.
If `bit_map!` gives the same result with faster speed, why thrown an error?